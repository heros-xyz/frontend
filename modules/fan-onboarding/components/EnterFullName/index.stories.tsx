import { Box } from "@chakra-ui/react";
import type { Meta, StoryObj } from "@storybook/react";
import EnterFullName from ".";

const meta: Meta<typeof EnterFullName> = {
  title: "Fan/On Boarding/EnterFullName",
  component: EnterFullName,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof EnterFullName>;

export const EnterFullNameComponent: Story = {
  render: (args) => (
    <Box h="100vh" bg="secondary">
      <EnterFullName {...args} />
    </Box>
  ),
};
