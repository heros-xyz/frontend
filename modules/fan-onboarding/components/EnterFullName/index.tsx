import { Box, Input, Text } from "@chakra-ui/react";
import { useFormik } from "formik";
import * as Yup from "yup";
import { FullNameFanOnBoardingIcon } from "@/components/svg/FullNameFanOnBoarding";
import { ArrowRight } from "@/components/svg/ArrowRight";
import ErrorMessage from "@/components/common/ErrorMessage";
import { isValidString } from "@/utils/functions";
import HerosOnboardingWrapperNew from "@/components/ui/HerosOnboardingWrapperNew";

interface initialValues {
  firstName: string;
  lastName: string;
}
interface IProp {
  onSubmit: (values: initialValues) => void;
  initialValues: initialValues;
}

const EnterFullName: React.FC<IProp> = ({ initialValues, onSubmit }) => {
  const validationSchema = Yup.object().shape({
    firstName: Yup.string()
      .max(20, "First name cannot exceeds 20 characters")
      .test(
        "invalid-firstname",
        "This field contains text only",
        (value: string | undefined) => {
          if (value) {
            return isValidString(value);
          }
          return true;
        }
      )
      .required("This is a required field"),
    lastName: Yup.string()
      .max(20, "Last name cannot exceeds 20 characters")
      .test(
        "invalid-lastname",
        "This field contains text only",
        (value: string | undefined) => {
          if (value) {
            return isValidString(value);
          }
          return true;
        }
      )
      .required("This is a required field"),
  });
  const formik = useFormik({
    initialValues,
    validationSchema: validationSchema,
    onSubmit: (values) => {
      onSubmit(values);
    },
  });
  return (
    <HerosOnboardingWrapperNew
      Icon={
        <FullNameFanOnBoardingIcon
          w={{ base: "72px", xl: "115px" }}
          h={{ base: "90px", xl: "144px" }}
          color={{ base: "#FFFAE8", xl: "#E2FF65" }}
        />
      }
      textButton="Proceed"
      IconButton={<ArrowRight />}
      onSubmit={formik.handleSubmit}
      bgIconColor="#E2FF65"
    >
      <Box color="primary">
        <Box mb={{ base: 5, lg: 8 }}>
          <Box mb={2.5} fontSize={{ lg: "xl" }} fontWeight="500">
            Enter Your Full Name
            <Text as="span" color="error.dark">
              {" "}
              *
            </Text>
          </Box>
          <Box
            fontWeight="normal"
            color="grey.300"
            fontSize={{ base: "xs", lg: "md" }}
          >
            Let&apos;s your favorite athletes know who is following and
            supporting them. Your full name will be shown to the public.
          </Box>
        </Box>
        <Box mb={4}>
          <Box mb={{ base: 5, lg: 8 }}>
            <Input
              id="firstName"
              variant="flushed"
              placeholder="First name"
              fontSize={{ base: "sm", lg: "3xl" }}
              value={formik.values?.firstName}
              onChange={formik.handleChange}
              isInvalid={Boolean(
                formik.errors.firstName && formik.touched.firstName
              )}
              borderColor="grey.200"
            />
            <ErrorMessage
              condition={formik.errors.firstName && formik.touched.firstName}
              errorMessage={formik.errors.firstName}
            />
          </Box>

          <Box>
            <Input
              id="lastName"
              variant="flushed"
              placeholder="Last name"
              fontSize={{ base: "sm", lg: "3xl" }}
              value={formik.values?.lastName}
              onChange={formik.handleChange}
              isInvalid={Boolean(
                formik.errors.lastName && formik.touched.lastName
              )}
              borderColor="grey.200"
            />
            <ErrorMessage
              condition={formik.errors.lastName && formik.touched.lastName}
              errorMessage={formik.errors.lastName}
            />
          </Box>
        </Box>
      </Box>
    </HerosOnboardingWrapperNew>
  );
};

export default EnterFullName;
