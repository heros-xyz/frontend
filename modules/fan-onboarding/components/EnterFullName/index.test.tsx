import { render } from "@testing-library/react";
import EnterFullName from ".";

it("renders homepage unchanged", () => {
  const onChangeStep = (step: number) => {};
  const { container } = render(
    <EnterFullName onSubmit={() => onChangeStep(2)} />
  );
  expect(container).toMatchSnapshot();
});
