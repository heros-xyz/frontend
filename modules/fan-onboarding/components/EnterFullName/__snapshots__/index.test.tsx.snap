// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="css-1jvt6v8"
  >
    <div
      class="css-1kmmdcq"
    />
    <div
      class="chakra-container css-17f3xev"
    >
      <div
        class="css-1ispe57"
      >
        <div
          class="css-rsp2ru"
        >
          <div
            class="css-1m8a01k"
          >
            <svg
              class="chakra-icon css-1ot4zdl"
              focusable="false"
              viewBox="0 0 72 90"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_5448_58240)"
              >
                <path
                  d="M16.0496 25.9568L14.5449 36.0195H18.8709L27.7111 34.0445L28.9336 24.1699L16.0496 25.9568Z"
                  fill="white"
                />
                <path
                  d="M16.0496 25.9568L14.5449 36.0195H18.8709L27.7111 34.0445L28.9336 24.1699L16.0496 25.9568Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M42.8521 24.0742L41.4414 33.1965L49.6232 38.4629L51.1279 28.4943L42.8521 24.0742Z"
                  fill="white"
                />
                <path
                  d="M42.8521 24.0742L41.4414 33.1965L49.6232 38.4629L51.1279 28.4943L42.8521 24.0742Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M46.3322 3.85547C46.2382 3.94951 45.0156 12.4135 45.0156 12.4135L52.5391 16.1752L54.1379 7.89936L46.3322 3.85547Z"
                  fill="white"
                />
                <path
                  d="M46.3322 3.85547C46.2382 3.94951 45.0156 12.4135 45.0156 12.4135L52.5391 16.1752L54.1379 7.89936L46.3322 3.85547Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M32.0373 12.4133L30.7207 22.476L41.9119 22.7581L43.7928 12.5073C43.7928 12.5073 34.8586 11.0967 32.0373 12.4133Z"
                  fill="white"
                />
                <path
                  d="M32.0373 12.4133L30.7207 22.476L41.9119 22.7581L43.7928 12.5073C43.7928 12.5073 34.8586 11.0967 32.0373 12.4133Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M19.3417 5.54855L18.1191 13.7304L30.8151 11.7554L31.9436 3.76172L19.3417 5.54855Z"
                  fill="white"
                />
                <path
                  d="M19.3417 5.54855L18.1191 13.7304L30.8151 11.7554L31.9436 3.76172L19.3417 5.54855Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M5.23439 11.2852L3.35352 21.63C3.35352 21.63 13.1341 25.956 14.6388 25.2037C16.2375 24.5453 16.7077 14.2946 16.7077 14.2946L5.23439 11.2852Z"
                  fill="white"
                />
                <path
                  d="M5.23439 11.2852L3.35352 21.63C3.35352 21.63 13.1341 25.956 14.6388 25.2037C16.2375 24.5453 16.7077 14.2946 16.7077 14.2946L5.23439 11.2852Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M63.6361 89.4347C66.7396 89.4347 70.8775 86.9895 70.4073 81.2529C70.1251 78.5256 67.0217 78.4315 67.0217 78.4315C67.0217 78.4315 71.2537 77.1149 70.6894 70.6259C70.3132 66.1118 66.7395 66.582 66.7395 66.582C66.7395 66.582 71.2537 64.513 71.7239 59.1525C72.1941 53.1337 67.3979 53.0397 67.3979 53.0397C73.4167 49.466 71.7239 41.8485 66.9276 40.0616C62.6016 38.4629 39.6549 38.4629 39.6549 38.4629C31.6612 39.2152 25.1721 39.8735 25.1721 39.8735L9.6549 40.6259L7.58594 89.9989C7.58594 89.9989 29.5922 89.9049 63.6361 89.4347Z"
                  fill="currentColor"
                />
                <path
                  d="M30.908 52.1002C29.6854 56.6143 29.7794 60.282 32.3186 64.0438C34.1995 66.8651 38.3374 66.5829 40.9707 64.4199C44.2622 61.7867 46.8954 55.4858 41.9111 44.2945C41.9111 44.2945 38.3374 34.8902 25.3594 40.7209"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M30.4375 58.8711C29.2149 62.5388 24.6068 63.6673 21.6914 63.4792"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M5.42188 10.627C5.42188 10.627 11.3466 15.4232 22.2557 14.0125C33.1648 12.5078 40.5943 7.14733 52.914 16.9279"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M3.54102 22.0996C3.54102 22.0996 9.65387 26.7078 20.3749 25.2971C31.1899 23.8864 38.5253 18.5259 50.7511 28.2125"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M18.6827 5.73633L13.9805 36.3946"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M32.6005 3.47852L28.5566 33.5726"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M45.955 3.29102L40.9707 33.197"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M54.5133 7.71139L49.9051 39.2161C49.9051 39.2161 41.5352 28.9653 27.5227 34.1377C13.4161 39.4042 2.50698 35.6424 1.09632 33.8556C0.908235 33.6675 6.64491 1.31641 6.64491 1.31641C6.64491 1.31641 13.0399 6.30073 23.8549 4.89007C34.764 3.38537 42.0995 -2.06917 54.5133 7.71139Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M50.7525 33.291L49.3418 42.8835"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M17.9298 5.45453L16.5192 14.3887L4.76367 10.5329L6.17433 1.88086L17.9298 5.45453Z"
                  fill="#424142"
                />
                <path
                  d="M17.9298 5.45453L16.5192 14.3887L4.76367 10.5329L6.17433 1.88086L17.9298 5.45453Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M30.1566 13.2598L29.4042 23.6986L15.6738 25.7676L17.8368 14.1062L30.1566 13.2598Z"
                  fill="#424142"
                />
                <path
                  d="M30.1566 13.2598L29.4042 23.6986L15.6738 25.7676L17.8368 14.1062L30.1566 13.2598Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M45.2038 2.44531L44.1694 11.6616L31.8496 11.1914L33.0722 2.72744L45.2038 2.44531Z"
                  fill="#424142"
                />
                <path
                  d="M45.2038 2.44531L44.1694 11.6616L31.8496 11.1914L33.0722 2.72744L45.2038 2.44531Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M53.0094 15.7046L51.0345 28.3065L43.1348 23.7924L45.2037 13.3535L53.0094 15.7046Z"
                  fill="#424142"
                />
                <path
                  d="M53.0094 15.7046L51.0345 28.3065L43.1348 23.7924L45.2037 13.3535L53.0094 15.7046Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M41.9114 23.6996L39.9364 33.3861L29.0273 33.2921L31.0023 22.9473L41.9114 23.6996Z"
                  fill="#424142"
                />
                <path
                  d="M41.9114 23.6996L39.9364 33.3861L29.0273 33.2921L31.0023 22.9473L41.9114 23.6996Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M15.2023 25.5797L12.9452 36.3948L6.17407 35.6424L1.0957 33.8556L3.16467 22.5703L15.2023 25.5797Z"
                  fill="#424142"
                />
                <path
                  d="M15.2023 25.5797L12.9452 36.3948L6.17407 35.6424L1.0957 33.8556L3.16467 22.5703L15.2023 25.5797Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M46.8016 83.6983C46.8016 81.3472 48.8706 79.5604 51.7859 79.0902C47.8361 78.5259 44.9207 77.5855 45.0148 73.0714C45.1088 69.5918 48.7765 68.0871 52.1621 67.3347C47.7421 67.2407 44.7327 66.2062 44.2624 62.3504C43.7922 58.5886 46.7076 56.0494 50.2812 55.109C47.0838 54.5447 43.9803 52.5698 44.0744 48.9021C44.2624 45.0463 48.6825 43.5416 51.6919 43.5416L62.0367 43.3535"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_5448_58240"
                >
                  <rect
                    fill="white"
                    height="90"
                    transform="translate(0.25)"
                    width="71.4734"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
        </div>
        <div
          class="css-14s5ckc"
        >
          <div
            class="css-1s2o82u"
          >
            <div
              class="css-1uoge8t"
            >
              <div
                class="css-1p9c8n6"
              >
                <div
                  class="css-78ozrq"
                >
                  <div
                    class="css-1kt4cil"
                  >
                    Enter Your Full Name
                    <span
                      class="chakra-text css-1b8yx6g"
                    >
                       
                      *
                    </span>
                  </div>
                  <div
                    class="css-309iqz"
                  >
                    Let's your favorite athletes know who is following and supporting them. Your full name will be shown to the public.
                  </div>
                </div>
                <div
                  class="css-1itv5e3"
                >
                  <div
                    class="css-78ozrq"
                  >
                    <input
                      class="chakra-input css-1gtqc92"
                      id="firstName"
                      placeholder="First name"
                      value=""
                    />
                  </div>
                  <div
                    class="css-0"
                  >
                    <input
                      class="chakra-input css-1gtqc92"
                      id="lastName"
                      placeholder="Last name"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                class="css-lrby83"
              >
                <button
                  class="chakra-button css-k6pd2s"
                  type="button"
                >
                  Proceed
                  <span
                    class="css-9n3bbc"
                  >
                    <svg
                      class="chakra-icon css-iz364c"
                      focusable="false"
                      viewBox="0 0 15 14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M1.25 7L13.75 7M13.75 7L8.125 1.375M13.75 7L8.125 12.625"
                        stroke="#33EFEF"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <button
            class="chakra-button css-1mv50kv"
            type="button"
          >
            Proceed
            <span
              class="css-9n3bbc"
            >
              <svg
                class="chakra-icon css-iz364c"
                focusable="false"
                viewBox="0 0 15 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1.25 7L13.75 7M13.75 7L8.125 1.375M13.75 7L8.125 12.625"
                  stroke="#33EFEF"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
